### ISSUE TITLE
Array changes are not always detected

### ISSUE THREAD

**Issue by mmccall10** [bitcrowd/carbonite#82]
Hi 👋 

When tracking changes for an array field it doesn't always detect a difference. I _think_ it's due to how the fields are compared `<@`.  

When an item is added to the list and the previous item is not removed the field is not detected as a change. 

```sql
-- NOT change_row.data @> old_field_jsonb

-- changed
SELECT NOT '["new"]'::jsonb @> '["old"]'::jsonb; 

-- changed
SELECT NOT '["new"]'::jsonb @> '["old", "new"]'::jsonb;

-- should be changed
SELECT NOT '["new", "old"]'::jsonb @> '["old"]'::jsonb;
```

**Comment by maltoe** [comment]
Hi @mmccall10 

thank you, this is in fact a bug, and a quite severe one as well :see_no_evil: Sorry for the inconvenience. We're working on a fix, shouldn't be too hard.

**Comment by maltoe** [comment]
First debugging:

What we're comparing is not the arrays directly with `@>` as in your example, but the `data` JSON object (= the entire `NEW` record) against an object of a single column and its value, e.g. `{"some_field": ["some_value"]}`. See the code [here](https://github.com/bitcrowd/carbonite/blob/main/lib/carbonite/migrations/v5.ex#L80). Just as a side-note, the issue comes out the same.

[Postgres docs](https://www.postgresql.org/docs/9.5/functions-json.html) about the `@>` operator:

> Does the left JSON value contain the right JSON path/value entries at the top level?

I guess it's the "at the top level" part that I have missed or didn't understand at the time.

```sql
psql=# SELECT '{"a": ["foo"]}'::jsonb @> '{"a": []}'::jsonb;
 ?column?
----------
 t
(1 row)

psql=# SELECT '{"a": []}'::jsonb @> '{"a": ["foo"]}'::jsonb;
 ?column?
----------
 f
(1 row)
```

It also doesn't work for objects.

```sql
psql=# SELECT '{"a": {"b": 1}}'::jsonb @> '{"a": {"b": 2}}'::jsonb;
 ?column?
----------
 f
(1 row)
```

Working on a fix.

**Comment by maltoe** [comment]
@mmccall10 Thanks again for noticing us. This is quite the embarrassing bug, and obviously very unfortunate for people who have relied on Carbonite correctly tracking all record versions. My apologies if this bug has already caused any damage to you.

**Comment by maltoe** [comment]
Fix released as v0.9.0

**Comment by mmccall10** [comment]
@maltoe Wow, that was fast! Very much appreciated 🙇‍♂️
